src/App.tsx

```diff
@@ -1,61 +1,135 @@
 import './App.scss';
+import React, { useState } from 'react';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+import { Todo } from './types/Todo';
+import { User } from './types/User';
+
+import { TodoList } from './components/TodoList';
+
+function findUser(id: number): User | null {
+  return usersFromServer.find(user => user.id === id) || null;
+}
+
+const todoList: Todo[] = todosFromServer.map(todo => (
+  {
+    ...todo,
+    user: findUser(todo.id),
+  }
+));
+
+function generateNextId(todos: Todo[]): number {
+  const idArray = todos.map(todo => todo.id).sort((a, b) => b - a);
+
+  return idArray[0] + 1;
+}
 
 export const App = () => {
+  const [todos, setTodos] = useState(todoList);
+  const [title, setTitle] = useState('');
+  const [userId, setUserId] = useState(0);
+  const [userError, setUserError] = useState(false);
+  const [titleError, setTitleError] = useState(false);
+
+  const clear = () => {
+    setTitle('');
+    setUserId(0);
+  };
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { value } = event.target;
+
+    setTitle(value);
+    setTitleError(false);
+  };
+
+  const handleUserChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    const { value } = event.target;
+
+    setUserId(+value);
+    setUserError(false);
+  };
+
+  const handleFormSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (!userId) {
+      setUserError(true);
+    }
+
+    if (!title) {
+      setTitleError(true);
+    }
+
+    if (title && userId) {
+      const newTodo = {
+        id: generateNextId(todos),
+        title,
+        completed: false,
+        userId,
+        user: findUser(userId),
+      };
+
+      setTodos(state => [...state, newTodo]);
+
+      clear();
+    }
```

Comment:
```suggestion
      const newTodo = {
        id: generateNextId(todos),
        title,
        completed: false,
        userId,
        user: findUser(userId),
      };

      setTodos(state => [...state, newTodo]);

      clearForm();
```

You don't need to check it twice